moveos/moveos/src/moveos.rs

```diff
@@ -243,12 +268,75 @@ impl MoveOS {
         }
     }
 
-    fn execute_gas_validate(
-        &self,
-        session: &mut MoveOSSession<'_, '_, MoveOSResolverProxy<MoveOSStore>, MoveOSGasMeter>,
-        action: &VerifiedMoveAction,
-    ) -> VMResult<Option<bool>> {
-        match action {
+    pub fn get_address_balance(&self, tx: &MoveOSTransaction) -> VMResult<u128> {
```

Comment:
MoveOs do not know the balance and do not know the `0x3::gas_coin`, so we should migrate this logic to the executor.

Reply:
OK, it's fixed.